app/backend/approaches/chatreadretrieveread.py

```diff
@@ -133,19 +137,52 @@ async def run_until_final_call(
             new_user_content=user_query_request,
             max_tokens=self.chatgpt_token_limit - query_response_token_limit,
         )
+        
+        if self.use_hugging_face:
+            huggingf_search_prompt = ""
+            
+            for message in query_messages:
+                huggingf_search_prompt += message["role"] + ": "
+                huggingf_search_prompt += message["content"] + '\n'
+                                
+            API_URL = os.gentenv("HUGGINGFACE_API_URL")
+            API_KEY = os.getenv("HUGGINGFACE_API_KEY")
+            headers = {"Authorization": "Bearer " + API_KEY}
+            
+            def ask_huggingface(payload):
+                response = requests.post(API_URL, headers=headers, json=payload)
+                return response.json()
+            
+            chat_completion = ask_huggingface({
+                "inputs": huggingf_search_prompt,
+                "parameters": {
+                    "temperature" : 1.3,
+                    "max_length": 400,
+                    "return_full_text": False,
+                    "top_k": 1,
+                    "top_p": 0.8,
+                    "repetition_penalty": 0.6,
+                    "max_new_tokens": 250,
+                },
+                "options": {
+                    "use_cache": False
+                }
```

Comment:
Hardcoding config like that here feels awkward. At the very least, this could be set in a constant easily accessible at the top of the file. This could also be part of the HF utils module btw.

Apart from that, should all of these be hardcoded? It's nice having the optimal configuration here as default, but would it also make sense exposing this to the user, e.g. allowing them to specify some of these values from the UI?